rules/PHPUnit120/Rector/Class_/AssertIsTypeMethodCallRector.php

```diff
@@ -0,0 +1,123 @@
+use Rector\PHPUnit\NodeAnalyzer\TestsNodeAnalyzer;
+use Rector\Rector\AbstractRector;
+use Symplify\RuleDocGenerator\ValueObject\CodeSample\CodeSample;
+use Symplify\RuleDocGenerator\ValueObject\RuleDefinition;
+
+/**
+ * @see https://github.com/sebastianbergmann/phpunit/issues/6053
+ * @see https://github.com/sebastianbergmann/phpunit/blob/12.0.0/ChangeLog-12.0.md
+ * @see \Rector\PHPUnit\Tests\PHPUnit120\Rector\MethodCall\AssertIsTypeMethodCallRector\AssertIsTypeMethodCallRectorTest
+ */
+final class AssertIsTypeMethodCallRector extends AbstractRector
+{
+    private const IS_TYPE_VALUE_TO_METHOD = [
+        'array' => 'isArray',
+        'bool' => 'isBool',
+        'boolean' => 'isBool',
+        'callable' => 'isCallable',
+        'double' => 'isFloat',
+        'float' => 'isFloat',
+        'integer' => 'isInt',
+        'int' => 'isInt',
+        'iterable' => 'isIterable',
+        'null' => 'isNull',
+        'numeric' => 'isNumeric',
+        'object' => 'isObject',
+        'real' => 'isFloat',
+        'resource' => 'isResource',
+        'resource (closed)' => 'isClosedResource',
+        'scalar' => 'isScalar',
+        'string' => 'isString',
+    ];
+
+    public function __construct(
+        private readonly ValueResolver $valueResolver,
+        private readonly TestsNodeAnalyzer $testsNodeAnalyzer,
+    ) {
+    }
+
+    public function getRuleDefinition(): RuleDefinition
+    {
+        return new RuleDefinition(
+            'Replaces `Assert::isType()` calls with type-specific `Assert::is*()` calls',
+            [
+                new CodeSample(
+                    <<<'CODE_SAMPLE'
+                    use PHPUnit\Framework\TestCase;
+
+                    final class SomeClass extends TestCase
+                    {
+                        public function testMethod(): void
+                        {
+                            $this->assertThat([], $this->isType('array'));
+                        }
+                    }
+                    CODE_SAMPLE
+                    ,
+                    <<<'CODE_SAMPLE'
+                    use PHPUnit\Framework\TestCase;
+
+                    final class SomeClass extends TestCase
+                    {
+                        public function testMethod(): void
+                        {
+                            $this->assertThat([], $this->isArray());
+                        }
+                    }
+                    CODE_SAMPLE
+                    ,
+                ),
+            ],
+        );
+    }
+
+    /**
+     * @return array<class-string<Node>>
+     */
+    public function getNodeTypes(): array
+    {
+        return [MethodCall::class, StaticCall::class];
+    }
+
+    /**
+     * @param MethodCall|StaticCall $node
+     */
+    public function refactor(Node $node): Node|null
+    {
+        if (! $this->testsNodeAnalyzer->isPHPUnitTestCaseCall($node) || ! $this->isName($node->name, 'isType')) {
+            return null;
+        }
+
```

Comment:
check first class callable first:

```suggestion
        if ($node->isFirstClassCallable()) {
            return null;
        }
```